benchmarks/benchmarks.c

```diff
@@ -55,3 +54,188 @@ void benchmarkFillup ()
 		}
 	}
 }
+
+#ifdef ELEKTRA_ENABLE_OPTIMIZATIONS
+
+// needed for the opmphmRandom ()
+#include "../src/libs/elektra/opmphm.c"
+
+/**
+ * @brief Fills a string up with random chars and null terminates it.
+ *
+ * @param name the string
+ * @param length the length of the string (excluding the '\0')
+ * @param seed to generate random data
+ *
+ */
+static void fillUpWithRandomChars (char * name, int length, uint32_t * seed)
+{
+	int i = 0;
+	char * c = name;
+	while (i < length)
+	{
+
+		if ((opmphmRandom (seed) % 10) == 0)
```

Comment:
I am not happy with the while function, I think it would be better to define somewhere a string with all possible chars in it and pick randomly from there.

The 32 bit limit for that type comes from the limitation of the `opmphmHashFunction ()` but the random numbers can surely be 64 bit.